src/hipBin_base.h

```diff
@@ -0,0 +1,869 @@
+    path = std::string(wide.begin(), wide.end());
+  #else
+    char buff[MAX_PATH_CHAR];
+    ssize_t len = ::readlink("/proc/self/exe", buff, sizeof(buff) - 1);
+    if (len != -1) {
+      buff[len] = '\0';
+      path = std::string(buff);
+      fs::path exePath(path);
+      path = exePath.parent_path().string();
+    }
+  #endif
+  return path;
+}
+
+
+// removes the empty spaces and end lines
+string HipBinBase::chomp(string str) const {
+  string strChomp = str;
+  strChomp.erase(str.find_last_not_of(" \n\r\t")+1);
+  return strChomp;
+}
+
+// matches the pattern in the string
+bool HipBinBase::stringRegexMatch(string fullString, string pattern) const {
+  bool found = false;
+  std::regex e(pattern);
+  if (std::regex_match(fullString, e)) {
+    found = true;
+  }
+  return found;
+}
+
+// subtring is present in string
+bool HipBinBase::substringPresent(string fullString, string subString) const {
+  bool found = false;
+  if (fullString.find(subString) != std::string::npos) {
+    found = true;
+  }
+  return found;
+}
+
+// splits the string with the delimiter
+vector<string> HipBinBase::splitStr(string fullStr, char delimiter) const {
+  vector <string> tokens;
+  stringstream check1(fullStr);
+  string intermediate;
+  while (getline(check1, intermediate, delimiter)) {
+    tokens.push_back(intermediate);
+  }
+  return tokens;
+}
+
+
+// replaces the toReplace string with replaceWith string. Returns the new string
+string HipBinBase::replaceStr(const string& s, const string& toReplace, const string& replaceWith) const {
+  string out = s;
+  std::size_t pos = out.find(toReplace);
+  if (pos == std::string::npos) return out;
+  return out.replace(pos, toReplace.length(), replaceWith);
+}
+
+// replaces the toReplace regex pattern with replaceWith string. Returns the new string
+string HipBinBase::replaceRegex(const string& s, std::regex toReplace, string replaceWith) const {
+  std::string out = s;
+  while (std::regex_search(out, toReplace)) {
+    out = std::regex_replace(out, toReplace, replaceWith);
+  }
+  return out;
+}
+
+// prints the envirnoment variables
+void HipBinBase::printEnvirnomentVariables() const {
+  const OsType& os = getOSInfo();
+  if (os == windows) {
+    cout << "PATH=" << envVariables_.path_ << "\n" << endl;
+    system("set | findstr /B /C:\"HIP\" /C:\"HSA\" /C:\"CUDA\" /C:\"LD_LIBRARY_PATH\"");
+  } else {
+    string cmd = "echo PATH =";
+    cmd += envVariables_.path_;
+    system(cmd.c_str());
+    system("env | egrep '^HIP|^HSA|^CUDA|^LD_LIBRARY_PATH'");
+  }
+}
+
+// prints system information
+void HipBinBase::getSystemInfo() const {
+  const OsType& os = getOSInfo();
+  if (os == windows) {
+    cout << endl << "== Windows Display Drivers" << endl;
+    cout << "Hostname      :";
+    system("hostname");
+    system("wmic path win32_VideoController get AdapterCompatibility,"
+    "InstalledDisplayDrivers,Name | findstr /B /C:\"Advanced Micro Devices\"");
+  } else {
```

Comment:
```suggestion
  } else {
    assert(os == linux);
```
Let's add this if this branch is meant to work in linux (compatible) OS.

Reply:
added the assert statement in the latest update